input/postgres/collection.go

```diff
@@ -56,5 +80,69 @@ func NewCollection(ctx context.Context, logger *util.Logger, server *state.Serve
 		Roles:                     roles,
 		ConnectedAsSuperUser:      connectedAsSuperUser,
 		ConnectedAsMonitoringRole: connectedAsMonitoringRole,
+		HelperFunctions:           helpersFromFunctions(helperFunctions),
 	}, nil
 }
+
+func (c *Collection) ForCurrentDatabase(functions []state.PostgresFunction) *Collection {
+	return &Collection{
+		Config:                    c.Config,
+		Logger:                    c.Logger,
+		SelfTest:                  c.SelfTest,
+		GlobalOpts:                c.GlobalOpts,
+		PostgresVersion:           c.PostgresVersion,
+		Roles:                     c.Roles,
+		ConnectedAsSuperUser:      c.ConnectedAsSuperUser,
+		ConnectedAsMonitoringRole: c.ConnectedAsMonitoringRole,
+		HelperFunctions:           helpersFromFunctions(functions),
+	}
+}
+
+func (c *Collection) findHelperFunction(name string, inputTypes []string) (state.PostgresFunction, bool) {
+	funcs, exists := c.HelperFunctions[name]
+	if !exists {
+		return state.PostgresFunction{}, false
+	}
+	for _, f := range funcs {
+		args := strings.Split(f.Arguments, ", ")
+		if len(inputTypes) > len(args) {
+			// We're expecting more arguments than the function has
+			continue
+		}
+		mismatch := false
+		for idx, arg := range args {
+			// Split by this assumed pattern: "NAME TYPE DEFAULT DEFAULT_VALUE"
+			//
+			// Note this currently does not handle data types with spaces in
+			// them, such as "double precision", or other cases not expected
+			// with our known set of helper functions.
+			parts := strings.Split(arg, " ")
+			if idx >= len(inputTypes) && parts[2] != "DEFAULT" {
```

Comment:
So extra arguments with a default are okay, right? Maybe we should add a comment to make that more obvious.